app/admin/top-issues/components/AdminTopIssuesPage.js

```diff
@@ -1,32 +1,35 @@
 'use client';
 import PortalPanel from '@shared/layouts/PortalPanel';
 import AdminWrapper from 'app/admin/shared/AdminWrapper';
-import Tooltip from '@mui/material/Tooltip';
-import Table from '@shared/utils/Table';
-import { useMemo, useState } from 'react';
-import { formatToPhone } from 'helpers/numberHelper';
-import gpApi, { isProd } from 'gpApi';
+import React, { useState } from 'react';
+import gpApi from 'gpApi';
 import gpFetch from 'gpApi/gpFetch';
-import AlertDialog from '@shared/utils/AlertDialog';
-import { FaTrash } from 'react-icons/fa';
-
 import { useHookstate } from '@hookstate/core';
 import { globalSnackbarState } from '@shared/utils/Snackbar';
 import BlackButtonClient from '@shared/buttons/BlackButtonClient';
 import TextField from '@shared/inputs/TextField';
 import TopIssuesList from './TopIssuesList';
+import { SVGIconChooser } from '@shared/buttons/SVGIconChooser';
 
-export const createTopIssue = async (name) => {
+export const createTopIssue = async (name, icon) => {
   const api = gpApi.admin.topIssues.create;
-  const payload = { name };
+  const payload = {
+    name,
+    ...(
+      icon ? {icon} : {}
+    )
+  };
```

Comment:
this is more readable
```
const payload = {
  name,
  icon: icon || {}
}
```

Reply:
That doesn't have the same functionality as the original code.

The former conditionally adds the key to the resultant object, the latter always adds an `icon` key, and in this case, would make it's value `{}`, which wouldn't be a valid value for that field since it's a `string | null` field? 🤔